app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,84 @@
+class RepositoriesController < ApplicationController
+  # def initialize
+  #   @github = Github.new
+  # end
+
+  def index
+    @repositories = Repository.all
+  end
+
+  def show
+    @repository = Repository.find(params[:id])
+    # @repository = response_from_uri("https://api.github.com/repos/NishantUpadhyay-BTC/code_forest")
+    @language = response_from_uri(@repository[:languages_url])
+    @total = @language.values.sum
+    @user = author_info('rails')
+  end
+
+  def new
+    repository_values_result = repository_values('NishantUpadhyay-BTC', 'code_forest')
+    @repository = Repository.new(repository_values_result[:repository_details])
+
+    repository_values_result[:language].each do |language,code|
+      @repository.languages.build(name: language, code: code)
+    end
+
+    impressionist(@repository, nil, { unique: [:session_hash] })
+  end
+
+  def create
+    @repository = Repository.new(repository_params)
+    save = @repository.save
+    redirect_to repositories_path
+  end
+
+  private
+  def response_from_uri(uri)
+    uri = URI(uri)
+    http_response = Net::HTTP.get_response(uri)
+    response = JSON(http_response.body).symbolize_keys
+    response
+  end
+
+  #returns authors infor from github id.
+  def author_info(user_name)
+    user_info = response_from_uri("https://api.github.com/users/#{user_name}")
+  end
+
+  #returns repository details for initializing @repository object in new action
+  #returns hash of languages used in that repository
+  def repository_values(user_name, repository_name)
```

Comment:
Implement Service Pattern. Move business logic to a service and call it in your controller method. Controller's job is to handle request and return response. Nothing else. http://blog.arkency.com/2013/09/services-what-they-are-and-why-we-need-them/